server/social.coffee

```diff
@@ -387,6 +387,75 @@ module.exports = exports = (log, loga, argv) ->
       }
       res.render(path.join(__dirname, '..', 'views', 'done.html'), info)
 
+    # if configured, enforce restricted access to json
+    if argv.restricted?
+
+      loginToView = (req) ->
+        # things = [ { value: 'ward.cunningham@gmail.com', type: 'account' } ]
+        allowed = [
```

Comment:
The list of allowed email domains could be loaded from a config item, not sure of the best name but something like `allowed_domains`. So config for the hardcoded example would be something like:

``` json
  "wikiDomains": {
    "example.wiki": {
      "restricted": true,
      "allowed_domains": [ "dayton.k12.or.us", "stu.dayton.k12.or.us",
          "newberg.k12.or.us", "stu.newberg.k12.or.us",
          "innovateoregon.org",
          "newrelic.com",
          "c2.com",
          "andysylvester.com"
      ]
    }
  }
```

So, line 395 would become:
``` coffeescript
        if argv.allowed_domains? and Array.isArray(argv.allowed_domains)
          allowed = argv.allowed_domains
        else
          allowed = []
```

Reply:
Well, this looks really simple. But how does allowed_domains get into argv?